hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1142,7 +1142,7 @@ def as_strategy(strat_or_callable, thing, final=True):
                     raise InvalidArgument(
                         f"`{k}: {v.__name__}` is not a valid type annotation"
                     ) from None
-            anns[k] = from_type(v)
+            anns[k] = _from_type(v, recurse_guard)
```

Comment:
It seems to be impossible to create a recursive constructible type without going through the class constructor args. I don't know if it really is, or that I just don't know how. My attempts all run into being unable to resolve the `ForwardRef` (`ResolutionFailed`), probably because the enclosing namespace is unavailable (except in the class `__init__` case).

But is this the canonical explanation to leave in a comment? I'm certainly not sure of my conclusion.

Reply:
I suspect `TypedDict` has special handling, since we use `st.fixed_dictionaries()` rather than creating an instance of the class?  But also unsure, tbh, I'd have to spend a while digging into the code again.